close #1 Add initial implementation for Telegram birthday bot

funaska opened the pull request.

funaska committed:
Add initial implementation for Telegram birthday bot

Includes core functionality for user birthday registration, birthday notifications, and customizable congratulation messages. Introduces backend database handling, bot command parsing, and admin capabilities with security checks. Relevant configurations added to `.gitignore`, and Python dependencies added to `requirements.txt`.

ValeryaKaminskaya commented:
- От имени админа не работают изменения текста поздравления, удаления дня рождения других пользователей
- После добавления меня как админа не могу изменить/удалить свое день рождения и текст поздравления
- Переделать регистрацию на обычные параметры (Просто чтобы не забыл😁)

funaska committed:
Fix pull request comments.

Updated `set_birthday` and related commands to support targeting specific users and handling user-specific chats. Added checks for user validity, chat membership, and improved exception handling. Enhanced `db_tools` with query updates and methods for improved birthday and congratulation management by chat.

funaska commented:
> * От имени админа не работают изменения текста поздравления, удаления дня рождения других пользователей
> * После добавления меня как админа не могу изменить/удалить свое день рождения и текст поздравления
> * Переделать регистрацию на обычные параметры (Просто чтобы не забыл😁)

Исправлены все пункты, можно тестировать заново

funaska committed:
Add handler for /congrat_text command

Implemented a new handler for the /congrat_text command to retrieve congratulatory messages for users. Admins can specify a target user ID, or it defaults to the command sender. Includes appropriate database call and fallback to the default congratulation message.

funaska commented:
Добавил команду /congrat_text для получения текста поздравления. Для администраторов чата или владельца бота есть возможность посмотреть текст для любого пользователя, передав user_id (по аналогии с другими командами).
Теперь бот делает почти всё из скоупа #1 , кроме расписания
